daemon/appinfo.c

```diff
@@ -0,0 +1,583 @@
+
+bool appinfo_parse_desktop(appinfo_t *self);
+
+/* ========================================================================= *
+ * APPINFO
+ * ========================================================================= */
+
+/* Ref: applications/jolla-email.desktop
+ *
+ * [Desktop Entry]
+ *
+ * Type=Application
+ * Name=Mail
+ * Icon=icon-launcher-email
+ * Exec=/usr/bin/sailjail -p jolla-email.desktop /usr/bin/jolla-email
+ *
+ * X-MeeGo-Logical-Id=mail-ap-name
+ * X-MeeGo-Translation-Catalog=jolla-email
+ * X-Maemo-Service=com.jolla.email.ui
+ * X-Maemo-Method=com.jolla.email.ui.activateWindow
+ * X-Maemo-Object-Path=/com/jolla/email/ui
+ * X-Desktop-File-Install-Version=0.26
+ *
+ * [X-Sailjail]
+ *
+ * Permissions=Accounts;Contacts;Phone;Email;WebView;Privileged;Internet;AppLaunch;Calendar
+ * OrganizationName=com.jolla
+ * ApplicationName=email
+ */
+
+struct appinfo_t
+{
+    // uplink
+    applications_t  *anf_applications;
+
+    gchar           *anf_appname;
+    appinfo_state_t  anf_state;
+    time_t           anf_dt_ctime;
+    bool             anf_dirty;
+
+    // read-only properties
+    gchar           *anf_dt_name;
+    gchar           *anf_dt_type;
+    gchar           *anf_dt_icon;
+    gchar           *anf_dt_exec;
+
+    stringset_t     *anf_sj_permissions_in;
+    stringset_t     *anf_sj_permissions_out;
+    gchar           *anf_sj_organization_name;
+    gchar           *anf_sj_application_name;
+};
+
+static void
+appinfo_ctor(appinfo_t *self, applications_t *applications, const gchar *id)
+{
+    self->anf_applications         = applications;
+    self->anf_appname              = g_strdup(id);
+
+    self->anf_state                = APPINFO_STATE_UNSET;
+    self->anf_dt_ctime             = -1;
+    self->anf_dirty                = false;
+
+    self->anf_dt_name              = NULL;
+    self->anf_dt_type              = NULL;
+    self->anf_dt_icon              = NULL;
+    self->anf_dt_exec              = NULL;
+
+    self->anf_sj_permissions_in    = stringset_create();
+    self->anf_sj_permissions_out   = stringset_create();
+    self->anf_sj_organization_name = NULL;
+    self->anf_sj_application_name  = NULL;
+
+    log_info("appinfo(%s): create", appinfo_id(self));
+}
+
+static void
+appinfo_dtor(appinfo_t *self)
+{
+    log_info("appinfo(%s): delete", appinfo_id(self));
+
+    appinfo_set_name(self, NULL);
+    appinfo_set_type(self, NULL);
+    appinfo_set_icon(self, NULL);
+    appinfo_set_exec(self, NULL);
+
+    stringset_delete_at(&self->anf_sj_permissions_in);
+    stringset_delete_at(&self->anf_sj_permissions_out);
+
+    change_string(&self->anf_appname, NULL);
+    self->anf_applications = NULL;
+}
+
+appinfo_t *
+appinfo_create(applications_t *applications, const gchar *id)
+{
+    appinfo_t *self = g_malloc0(sizeof *self);
+    appinfo_ctor(self, applications, id);
+    return self;
+}
+
+void
+appinfo_delete(appinfo_t *self)
+{
+    if( self != 0 ) {
```

Comment:
I like this better coding style overall better than what we have for sailjail command currently, so I would keep this. However I really never understood why to put spaces like this in conditions (no space after "if" but there are spaces inside the braces).

Also "for" is written in this fashion but "while" is not which feels strange to me.

Not a big deal, I just want to have some common understanding when going forward as multiple people will be contributing to the code.

Reply:
The idea is: language itself forces parentheses for conditionals. If  the used expression too needs parentheses, it can become burdensome to parse it visually. So, instead of

    if ((bar() < 7) && !(foo = something()))

I use

    if( (bar() < 7) && !(foo = something()) )

which I find much easier on my eyes and brain.

All conditionals [if/while/assert/...] ought to be like that... After git grep I'm guessing:  glib hash table iteration never enters my backbone -> copy paste those three lines from dev doc example & have a think-o.